server/controllers/tripController.js

```diff
@@ -0,0 +1,152 @@
+          status: status1
+        };
+        const { returnDate, destination } = body;
+        const cityNumber = destination.length;
+
+        newTrip = await TripController.setTripType(newTrip, returnDate, cityNumber);
+        const foundTrip = await tripHelpers.reasonsDate({
+          userId: myuserId,
+          reasons: body.reasons,
+          destination: body.destination,
+          date: body.date
+        });
+        if (foundTrip) {
+          return res.status(409).json({
+            status: 409,
+            data: 'You create this trip'
+          });
+        }
+        const saveTrip = await tripHelpers.saveTrip(newTrip);
+
+        return res.status(201).json({
+          status: 201,
+          message: 'Trip was created successfully.',
+          data: saveTrip
+        });
+      }
+      return res.status(401).json({
+        status: 401,
+        error: 'Only user with who have managers can make a trip request'
+      });
+    } catch (error) {
+      return res.status(500).json({
+        status: 500,
+        message: 'something goes wrong ',
+        error: error.message
+      });
+    }
+  }
+
+  /**
+   * This method handles view all trip requests.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @param {function} next The next action.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async viewAllTrips(req, res, next) {
+    try {
+      const { role, id } = req.user;
+      const {
+        start, end, pages, skip, paginate
+      } = await pagination.paginateData(req.query);
+      const foundTrips = await tripHelpers.findTripByRole(role, id, skip, start);
+      const userAllData = foundTrips.rows;
+      const countUserData = foundTrips.count;
+      if (foundTrips.rows.length === 0) {
+        return res.status(404).json({
+          status: 404,
+          message: `${req.user.email} You don't have trip request`
+        });
+      }
+      req.data = {
+        userAllData, countUserData, start, end, pages, skip, paginate
+      };
+      next();
+    } catch (error) {
+      return res.status(500).json({
+        status: 500,
+        message: ' something goes wrong ',
+        error: error.message
+      });
+    }
+  }
+
+  /**
+   * This method handles all update_trip requests.
+   * @param {object} req The user's request.
+   * @param {object} res The response.
+   * @returns {object} The status and some data of the trip.
+   */
+  static async viewOneTrip(req, res) {
+    try {
+      const { tripId } = req.params;
+      const { role, id } = req.user;
+      const foundTrip = await tripHelpers.findTripById(tripId, { role, id });
+      if (foundTrip) {
+        return res.status(200).json({
+          status: 200,
+          message: 'Single trip retrieved successfully',
+          data: foundTrip
+        });
+      }
+      res.status(401).json({
+        status: 401,
+        error: 'Unauthorized'
```

Comment:
Change the message to **You don’t have the permission for retrieving a trip request**